Add thread level details to _tasks API (#4542)
opensearch-project/documentation-website: _api-reference/tasks.md

```

### The `resource_stats` object

> The `resource_stats` object is only updated for tasks that support resource tracking. These stats are computed based on scheduled thread executions, which includes both threads that have finished working on the task and threads currently working on the task. Because the same thread may be scheduled to work on the same task multiple times, each time a given thread is scheduled to work on a given task is considered a single thread execution.

Reviewer: ```suggestion
The `resource_stats` object is only updated for tasks that support resource tracking. These stats are computed based on scheduled thread executions, including both threads that have finished working on the task and threads currently working on the task. Because the same thread may be scheduled to work on the same task multiple times, each instance of a given thread being scheduled to work on a given task is considered to be a single thread execution.
```

The following table lists all response fields in the `resource_stats` object. 

Response field | Description |
:--- | :--- |
`average` | The average resource usage across all scheduled thread executions. |
`total` | The sum of resource usages across all scheduled thread executions. |
`min` | The minimum resource usage across all scheduled thread executions. |
`max` | The maximum resource usage across all scheduled thread executions. |

> `thread_info` | Thread count-related stats.|

Reviewer: If this means "count-related stats for the thread", then the hyphenation is correct, but confirm that it shouldn't be "thread-count-related stats".

Reviewer 2: It is "thread-count-related". I'll change.

```

Note that `resource_stats` values will only be updated for tasks that support resource tracking. Moreover, `active_threads` refers to the number of threads currently working on the task while `thread_executions` refers to the number of threads that have been scheduled to work on the task, which includes both threads that have finished working on the task and threads currently working on the task.

> 

Reviewer 3: Can we add a table for defining all the stats emitted in `resource_stats` section. Like what does `min/max/average/total` mean here for a task similar to how you have defined `thread_info` section

Author: @sohami Thanks for the callout, added an additional table in the most recent revision.
